book/src/07_join.md

```diff
@@ -0,0 +1,130 @@
+we worked out a way to conveniently specify that. And this concept is
+known as "Joining".
+
+## Basic joining
+
+We've already seen some basic examples in the last chapters, for
+example how to join over two storages:
+
+```rust,ignore
+for (pos, vel) in (&mut pos_storage, &vel_storage).join() {
+    *pos += *vel;
+}
+```
+
+This simply iterates over the position and velocity components of
+all entities that have both these components. That means all the
+specified components are **required**.
+
+Sometimes, you want to not only get the components of entities,
+but also the entity itself. To do that, we can simply join over
+`&EntitiesRes`. We can get `&EntitiesRes` by de-referencing `Entities`
+and re-referencing the returned value.
+
+```rust,ignore
+for (ent, pos, vel) in (&*entities, &mut pos_storage, &vel_storage).join() {
+    println!("Processing entitiy: {:?}", ent);
+    *pos += *vel;
+}
+```
+
+## Optional components
+
+If we iterate over the `&EntitiesRes` as shown above, we can simply
+use the returned `Entitiy` values to get components from storages as usual.
+
+```rust,ignore
+for (ent, pos, vel) in (&*entities, &mut pos_storage, &vel_storage).join() {
+    println!("Processing entitiy: {:?}", ent);
+    *pos += *vel;
+    
+    let mass: Option<&mut Mass> = mass_storage.get_mut(ent);
+    if let Some(mass) = mass {
+        let x = *vel / 300_000_000.0;
+        let y = 1 - x * x;
+        let y = y.sqrt();
+        mass.current = mass.constant / y;
+    }
+}
+```
+
+In this example we iterate over all entities with a position and a velocity
+and perform the calculation for the new position as usual.
+However, in case the entity has a mass, we also calculate the current
+mass based on the velocity. Thus, mass is an **optional** component here.
+
+## Excluding components
+
+If you want to filter your selection by excluding all entities
+with a certain component type, you can use the not operator (`!`)
+on respective component storage. Its return value is a unit (`()`).
+
+```rust,ignore
+for (ent, pos, vel, ()) in (
+    &*entities,
+    &mut pos_storage,
+    &vel_storage,
+    !&freezed_storage,
+).join() {
+    println!("Processing entitiy: {:?}", ent);
+    *pos += *vel;
+}
+```
+
+This will simply iterate over all entities that
+
+* have a position
+* have a velocity
+* do not have a `Freezed` component
+
+## How joining works
+
+You can call `join()` on everything that implements the `Join` trait.
+The method call always returns an iterator. `Join` is implemented for
+
+* `&ReadStorage` / `&WriteStorage` (gives back a reference to the components)
+* `&mut WriteStorage` (gives back a mutable reference to the components)
+* `&EntitiesRes` (returns `Entity` values)
+* bitsets
+
+The last point here is probably very interesting, because
+this allows for even more flexibility, as you will see in the next
+section.
+
+## Joining over bitsets
+
+Specs is using `hibitset`, a library which provides layered bitsets
```

Comment:
s/is using/uses/

Reply:
My feeling tells me that I should use the progressive form since it's not something Specs does regularly. @Rhuagh @Xaeroxe is my feeling wrong?